src/drivers/spektrum_rc/spektrum_rc.cpp

```diff
@@ -0,0 +1,281 @@
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+
+/**
+ * @file spektrum_rc.cpp
+ *
+ * This is a driver for a Spektrum satellite receiver connected to a Snapdragon
+ * on the serial port. By default port J15 (next to USB) is used.
+ */
+
+#include <px4_tasks.h>
+#include <px4_posix.h>
+
+#include <lib/rc/dsm.h>
+#include <drivers/drv_rc_input.h>
+#include <drivers/drv_hrt.h>
+
+#include <uORB/uORB.h>
+#include <uORB/topics/input_rc.h>
+
+// Snapdraogon: use J15 (next to USB)
+#define SPEKTRUM_UART_DEVICE_PATH "/dev/tty-1"
+
+#define UNUSED(x) (void)(x)
+
+extern "C" { __EXPORT int spektrum_rc_main(int argc, char *argv[]); }
+
+
+namespace spektrum_rc
+{
+
+volatile bool _task_should_exit = false;
+static bool _is_running = false;
+static px4_task_t _task_handle = -1;
+
+int start();
+int stop();
+int info();
+void usage();
+void task_main(int argc, char *argv[]);
+
+void fill_input_rc(uint16_t raw_rc_count, uint16_t raw_rc_values[input_rc_s::RC_INPUT_MAX_CHANNELS],
+		   hrt_abstime now, bool frame_drop, bool failsafe, unsigned frame_drops, int rssi,
+		   input_rc_s &input_rc);
+
+void task_main(int argc, char *argv[])
+{
+	int uart_fd = dsm_init(SPEKTRUM_UART_DEVICE_PATH);
+
+	if (uart_fd < 1) {
+		PX4_ERR("dsm init failed");
+		return;
+	}
+
+	orb_advert_t rc_pub = nullptr;
+
+	// Use a buffer size of the double of the minimum, just to be safe.
+	uint8_t rx_buf[2 * DSM_BUFFER_SIZE];
+
+	_is_running = true;
+
+	// Main loop
+	while (!_task_should_exit) {
+
+		// sleep since no poll
+		usleep(10000);
+		int newbytes = ::read(uart_fd, &rx_buf[0], sizeof(rx_buf));
+
+		if (newbytes < 0) {
+			PX4_WARN("read failed");
+			continue;
+		}
+
+		if (newbytes == 0) {
+			PX4_WARN("read nothing");
```

Comment:
@Stifael You said this happens fairly often, so I'd remove the `PX4_WARN`. We don't want to spam the console and logs (`PX4_WARN`) is also logged in logger.